src/Lykke.Service.B2c2Adapter/Services/OrderBooksService.cs

```diff
@@ -192,9 +272,38 @@ private static IEnumerable<OrderBookItem> GetOrderBookItems(IEnumerable<Quantity
             return result;
         }
 
+        private void SetHelthCheck(PriceMessage result)
+        {
+            if (_healthCheck.TryGetValue(result.Instrument, out var counter))
+                _healthCheck[result.Instrument] = counter + 1;
+            else
+                _healthCheck[result.Instrument] = 1;
+        }
+
+        private Task WriteHealthCheck()
+        {
+            var list = _healthCheck.OrderBy(x => x.Value)
+                                   .Select(x => x.Key)
+                                   .Select(key => $"{key} : {_healthCheck[key]}")
+                                   .ToList();
+            try
+            {
+                _log.Info($"Health check: {Environment.NewLine}{string.Join(Environment.NewLine, list)}");
+            }
+            catch (Exception ex)
+            {
+                _log.Error(ex);
+            }
+
+            foreach (var key in _healthCheck.Keys)
+                _healthCheck[key] = 0;
```

Comment:
Shouldn't you use TryUpdate ? What if key is alread deleted?

Reply:
It will create a new one if not found, or may be I didn't understand your point.